smf/clickhouse/method_script.sh

```diff
@@ -28,13 +28,90 @@ ipadm show-addr "$DATALINK/ll" || ipadm create-addr -t -T addrconf "$DATALINK/ll
 ipadm show-addr "$DATALINK/omicron6"  || ipadm create-addr -t -T static -a "$LISTEN_ADDR" "$DATALINK/omicron6"
 route get -inet6 default -inet6 "$GATEWAY" || route add -inet6 default -inet6 "$GATEWAY"
 
-args=(
-  "--log-file" "/var/tmp/clickhouse-server.log"
-  "--errorlog-file" "/var/tmp/clickhouse-server.errlog"
-  "--"
-  "--path" "${DATASTORE}"
-  "--listen_host" "$LISTEN_ADDR"
-  "--http_port" "$LISTEN_PORT"
-)
-
-exec /opt/oxide/clickhouse/clickhouse server "${args[@]}" &
+# Retrieve addresses of the clickhouse nodes.
+CH_ADDRS="$(/opt/oxide/internal-dns-cli/bin/dnswait clickhouse -H \
+    | head -n 2 \
+    | tr '\n' ,)"
+
+if [[ -z "$CH_ADDRS" ]]; then
+    printf 'ERROR: found no addresses for other ClickHouse nodes\n' >&2
+    exit "$SMF_EXIT_ERR_CONFIG"
+fi
+
+readarray -td, nodes <<<"$CH_ADDRS,"; declare -p nodes
+
+for i in {0..1}
+do
+  if ! grep -q "host.control-plane.oxide.internal" <<< "${nodes["${i}"]}"; then
+    printf 'ERROR: retrieved ClickHouse address does not match the expected format\n' >&2
+    exit "$SMF_EXIT_ERR_CONFIG"
+  fi
+done
+
+# Assign addresses to replicas
```

Comment:
I think it would help if we made clear these are _hostnames_, specifically SRV records in internal DNS.